### PR TITLE
chore: 🤖 Add build admin UI workflow

### PR DESCRIPTION
**Pull request by calcaide**
Add build admin UI github actions workflow.

**Assumption**
This github action triggers after merging with main. It assumes before merging, the worflow `validate-monorepo` (compliance, lint and tests) has run. So no need to run again such workflow, but open to discussion if you think otherwise.

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-3849)

### CODE REVIEW

File: .github/workflows/build-admin-ui.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/cache@v3
+        id: node-modules-cache
+        with:
+          path: '**/node_modules'
+          key: node-modules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            node-modules-
+      - uses: actions/setup-node@v3
```
Comment (DhariniJeeva) [inline .github/workflows/build-admin-ui.yaml:20]: this is not a blocker and can be addressed later. I wonder if we should update our setup-node version [here](https://github.com/hashicorp/boundary-ui/blob/main/.github/workflows/monorepo-validate.yaml#L55) as well?

**Comment by DhariniJeeva** [comment]
LGTM!

**Review by DhariniJeeva — APPROVED** [review]

File: .github/workflows/build-admin-ui.yaml (hunk 1 above)
Comment (calcaide) [reply .github/workflows/build-admin-ui.yaml:20]: @DhariniJeeva You got it! Nice catch! It was on the toDo list but you are right, there is no reason for it to be included on this PR.
I changed to the V3 in all the actions, they release it when I started testing this 

Nice catch!

**Review by cameronperera — APPROVED** [review]
LGTM

**Pull request merged**